app.js

```diff
@@ -1,11 +1,125 @@
 /* Imports */
 // this will check if we have a user and set signout link if it exists
 import './auth/user.js';
+import { createItem, getItem, buyItem, deleteItem, deleteBought } from './fetch-utils.js';
 
 /* Get DOM Elements */
+const addItemForm = document.getElementById('add-item-form');
+const itemList = document.getElementById('item-list');
+const errorDisplay = document.getElementById('error-display');
+const removeButton = document.getElementById('remove-button');
+const removeBought = document.getElementById('remove-bought');
 
 /* State */
+let items = [];
+let error = null;
 
 /* Events */
 
+window.addEventListener('load', async () => {
+    const response = await getItem();
+    error = response.error;
+    items = response.data;
+
+    if (error) {
+        displayError();
+    }
+
+    if (items) {
+        displayItems();
+    }
+});
+
+addItemForm.addEventListener('submit', async (e) => {
+    e.preventDefault();
+    const formData = new FormData(addItemForm);
+    const newItem = {
+        item: formData.get('item'),
+        quantity: formData.get('quantity'),
+    };
+
+    const response = await createItem(newItem);
+    error = response.error;
+    const item = response.data;
+
+    if (error) {
+        displayError();
+    } else {
+        items.push(item);
+        displayItems();
+        addItemForm.reset();
+    }
+});
+
+removeButton.addEventListener('click', async () => {
+    const response = await deleteItem();
+    error = response.error;
+
+    if (error) {
+        displayError();
+    } else {
+        items = [];
+        displayItems();
+    }
+});
+
+removeBought.addEventListener('click', async () => {
+    const response = await deleteBought();
+    error = response.error;
+
+    if (error) {
+        displayError();
+    } else {
+        items = [];
```

Comment:
The `deleteBought()` function works on the backend, but then you are setting `items` to an empty array.
If `items` is what you are iterating through to display your list, wouldn't you want to push the items that persist in your Supabase table into the `items` array here?